### ISSUE TITLE
Star unpacking in class arguments doesn't mark variables as used

### ISSUE THREAD

**Issue by Bobronium** [microsoft/pyright#3895]
<h3> Environment data </h3>

-   Language Server version: 2022.8.50
-   OS and version: darwin arm64
-   Python version (& distribution if applicable, e.g. Anaconda): 3.10.0, pyenv
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: basic

## Code Snippet

```python
def create_subclass(*base_classes, **kwargs):
    class Subclass(*base_classes, **kwargs): ...
    return Subclass
```
<img width="459" alt="image" src="https://user-images.githubusercontent.com/36469655/188025114-b1347f69-5685-4001-9805-63dfeefce8ef.png">
<img width="639" alt="image" src="https://user-images.githubusercontent.com/36469655/188025217-faf64a45-93c7-4f0e-bd36-5606e44e06fd.png">

<img width="715" alt="image" src="https://user-images.githubusercontent.com/36469655/188025151-fad864a5-2089-4f15-833d-042c376f9ebd.png">

## Repro Steps

1. declare a variable `base_classes` in local scope
2. use that variable in `class Name(*base_classes): ...` expression and nowhere else

## Expected behavior

Variable is marked as used

## Actual behavior

Variable is grayed out and pylance complains that `"base_classes" is not accessed`

## Logs

<details>

```
[Info  - 2:32:53 AM] (13063) Pylance language server 2022.8.50 (pyright f107f0f3) starting
[Info  - 2:32:53 AM] (13063) Server root directory: /Users/rocky/.vscode/extensions/ms-python.vscode-pylance-2022.8.50/dist
[Info  - 2:32:53 AM] (13063) Notebook support: LSP
[Info  - 2:32:53 AM] (13063) Interactive window support: LSP
[Info  - 2:33:06 AM] (13063) Starting service instance "<default>"
[Info  - 2:33:07 AM] (13063) Background analysis(1) root directory: /Users/rocky/.vscode/extensions/ms-python.vscode-pylance-2022.8.50/dist
[Info  - 2:33:07 AM] (13063) Background analysis(1) started
[Info  - 2:33:07 AM] (13063) No pyproject.toml file found.
[Info  - 2:33:07 AM] (13063) Setting pythonPath for service "<default>": "/Users/rocky/.pyenv/versions/3.10.0/bin/python"
[Warn  - 2:33:07 AM] (13063) stubPath typings is not a valid directory.
[Info  - 2:33:07 AM] (13063) Assuming Python version 3.10
[Info  - 2:33:07 AM] (13063) Assuming Python platform Darwin
[Info  - 2:33:07 AM] (13063) Search paths for <default>
[Info  - 2:33:07 AM] (13063)   /Users/rocky/.vscode/extensions/ms-python.vscode-pylance-2022.8.50/dist/typeshed-fallback/stdlib
[Info  - 2:33:07 AM] (13063)   typings
[Info  - 2:33:07 AM] (13063)   /Users/rocky/.vscode/extensions/ms-python.vscode-pylance-2022.8.50/dist/typeshed-fallback/stubs/...
[Info  - 2:33:07 AM] (13063)   /Users/rocky/.vscode/extensions/ms-python.vscode-pylance-2022.8.50/dist/bundled/stubs
[Info  - 2:33:07 AM] (13063)   /Users/rocky/.pyenv/versions/3.10.0/lib/python3.10
[Info  - 2:33:07 AM] (13063)   /Users/rocky/.pyenv/versions/3.10.0/lib/python3.10/lib-dynload
[Info  - 2:33:07 AM] (13063)   /Users/rocky/.pyenv/versions/3.10.0/lib/python3.10/site-packages
[Info  - 2:33:07 AM] (13063) Searching for source files
[Info  - 2:33:07 AM] (13063) No source files found.
(13063) [IDX(FG)] index libraries  (index) ...
(13063) [IDX(FG)]   read stdlib indices (21ms)
(13063) [IDX(FG)] index libraries  (index) [succeed] (22ms)
(13063) [FG] parsing: Untitled-1 (11ms)
(13063) [FG] parsing: /Users/rocky/.vscode/extensions/ms-python.vscode-pylance-2022.8.50/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 2ms] (78ms)
(13063) [FG] binding: /Users/rocky/.vscode/extensions/ms-python.vscode-pylance-2022.8.50/dist/typeshed-fallback/stdlib/builtins.pyi (17ms)
(13063) [FG] binding: Untitled-1 (1ms)
(13063) Background analysis message: setConfigOptions
(13063) Background analysis message: setImportResolver
(13063) Background analysis message: ensurePartialStubPackages
(13063) Background analysis message: setTrackedFiles
(13063) Background analysis message: markAllFilesDirty
(13063) Background analysis message: setFileOpened
(13063) Background analysis message: getSemanticTokens full
(13063) [BG(1)] getSemanticTokens full at Untitled-1 ...
(13063) [BG(1)]   parsing: Untitled-1 (9ms)
(13063) [BG(1)]   parsing: /Users/rocky/.vscode/extensions/ms-python.vscode-pylance-2022.8.50/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 2ms] (79ms)
(13063) [BG(1)]   binding: /Users/rocky/.vscode/extensions/ms-python.vscode-pylance-2022.8.50/dist/typeshed-fallback/stdlib/builtins.pyi (25ms)
(13063) [BG(1)]   binding: Untitled-1 (0ms)
(13063) [BG(1)] getSemanticTokens full at Untitled-1 (114ms)
(13063) Background analysis message: analyze
(13063) [BG(1)] analyzing: Untitled-1 ...
(13063) [BG(1)]   checking: Untitled-1 (1ms)
(13063) [BG(1)] analyzing: Untitled-1 (1ms)
(13063) Background analysis message: resumeAnalysis
(13063) Background analysis message: setFileOpened
(13063) Background analysis message: markFilesDirty
(13063) [FG] parsing: Untitled-1 (1ms)
(13063) [FG] binding: Untitled-1 (0ms)
(13063) Background analysis message: analyze
(13063) [BG(1)] analyzing: Untitled-1 ...
(13063) [BG(1)]   parsing: Untitled-1 (1ms)
(13063) [BG(1)]   binding: Untitled-1 (0ms)
(13063) [BG(1)]   checking: Untitled-1 ...
(13063) [BG(1)]     parsing: /Users/rocky/.vscode/extensions/ms-python.vscode-pylance-2022.8.50/dist/typeshed-fallback/stdlib/typing.pyi [fs read 2ms] (20ms)
(13063) [BG(1)]     binding: /Users/rocky/.vscode/extensions/ms-python.vscode-pylance-2022.8.50/dist/typeshed-fallback/stdlib/typing.pyi (4ms)
(13063) [BG(1)]     parsing: /Users/rocky/.vscode/extensions/ms-python.vscode-pylance-2022.8.50/dist/typeshed-fallback/stdlib/typing_extensions.pyi [fs read 1ms] (3ms)
(13063) [BG(1)]     binding: /Users/rocky/.vscode/extensions/ms-python.vscode-pylance-2022.8.50/dist/typeshed-fallback/stdlib/typing_extensions.pyi (1ms)
(13063) IntelliCode model /Users/rocky/.vscode/extensions/visualstudioexptteam.vscodeintellicode-1.2.24/cache/E61945A9A512ED5E1A3EE3F1A2365B88F8FE_E4E9EADA96734F01970E616FAB2FAC19
(13063) [BG(1)]     parsing: /Users/rocky/.vscode/extensions/ms-python.vscode-pylance-2022.8.50/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi [fs read 0ms] (9ms)
(13063) [BG(1)]     binding: /Users/rocky/.vscode/extensions/ms-python.vscode-pylance-2022.8.50/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi (1ms)
(13063) [BG(1)]     parsing: /Users/rocky/.vscode/extensions/ms-python.vscode-pylance-2022.8.50/dist/typeshed-fallback/stdlib/types.pyi [fs read 0ms] (8ms)
(13063) [BG(1)]     binding: /Users/rocky/.vscode/extensions/ms-python.vscode-pylance-2022.8.50/dist/typeshed-fallback/stdlib/types.pyi (2ms)
(13063) [BG(1)]     parsing: /Users/rocky/.vscode/extensions/ms-python.vscode-pylance-2022.8.50/dist/typeshed-fallback/stdlib/abc.pyi [fs read 0ms] (1ms)
(13063) [BG(1)]     binding: /Users/rocky/.vscode/extensions/ms-python.vscode-pylance-2022.8.50/dist/typeshed-fallback/stdlib/abc.pyi (0ms)
(13063) [BG(1)]   checking: Untitled-1 (71ms)
(13063) [BG(1)] analyzing: Untitled-1 (72ms)
(13063) Background analysis message: resumeAnalysis
(13063) Background analysis message: getSemanticTokens delta
(13063) [BG(1)] getSemanticTokens delta previousResultId:1662071587509 at Untitled-1 (1ms)
(13063) Loading ONNX runtime...
(13063) Loaded ONNX runtime. Creating IntelliCode session...
(13063) Created IntelliCode session.
(13063) Initialize deeplearning succeeded
(13063) Background analysis message: setFileOpened
(13063) Background analysis message: markFilesDirty
(13063) [FG] parsing: Untitled-1 (1ms)
(13063) [FG] binding: Untitled-1 (0ms)
(13063) Background analysis message: analyze
(13063) [BG(1)] analyzing: Untitled-1 ...
(13063) [BG(1)]   parsing: Untitled-1 (2ms)
(13063) [BG(1)]   binding: Untitled-1 (0ms)
(13063) [BG(1)]   checking: Untitled-1 (17ms)
(13063) [BG(1)] analyzing: Untitled-1 (20ms)
(13063) Background analysis message: resumeAnalysis
(13063) Background analysis message: getSemanticTokens delta
(13063) [BG(1)] getSemanticTokens delta previousResultId:1662071588200 at Untitled-1 (1ms)
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version of pyright and a future version of pylance.

Commit is here: https://github.com/microsoft/pyright/commit/660e9945214c4d61427b8e7154179b30b8a037a1

**Comment by Bobronium** [comment]
Fastest bugfix in the west! Thank you.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.270, which I just published. It will also be included in a future release of pylance.
